components/selection/selection-mixin.js

```diff
@@ -223,7 +226,24 @@ export const SelectionMixin = superclass => class extends RtlMixin(CollectionMix
 			const info = this.getSelectionInfo(true);
 			this._selectionObservers.forEach(observer => observer.selectionInfo = info);
 			this._updateObserversRequested = false;
-		}, 0);
+		});
+	}
+
+	_updateSortedSelectables() {
+
+		if (this._updateSortedSelectablesRequested) return;
+
+		this._updateSortedSelectablesRequested = true;
+		setTimeout(() => {
+			this._sortedSelectables = Array.from(this._selectionSelectables.values())
+				.sort((a, b) => {
+					const idx = a.findIndex((el, idx) => el !== b[idx]);
+					return a[idx].compareDocumentPosition(b[idx]) & Node.DOCUMENT_POSITION_FOLLOWING ? -1 : 1;
```

Comment:
Yep, it just fails (or, more accurately, returns inconsistent results) across shadowDOM boundaries so you can only compare things in the same root.

Reply:
Does that shadowDOM limitation not cause us issues in list, since the `d2l-selection-input` elements are baked into separate DOM sub-trees of each list-item?

Reply:
Using `compareDocumentPosition` on the two selectables directly does cause issues, but this is comparing their common-root ancestors instead, using the composedPath we already have from the the subscribe event handler.